### DISCUSSION TITLE
Reflow text

### DISCUSSION THREAD

**Discussion by riksoft** [apache/netbeans#5961] [Ideas]
It would be nice to have a simple function to reflow the selected text, the equivalent of what the Geany Editor does pressing CTRL+J or with a Linux command like fmt.
Any editor should have this function. Maybe it's already in Netbeans and I don't find it? At least... any plugin?

That's quite useful especially when making websites and pasting text from another source.

**Comment by Chris2011** [comment]
Can ou please tell us more about this feature or make a little screencast? I don't know this feature also not the editor.

**Comment by riksoft** [comment]
Sure, here an example with the editor Geany editing the home page of netbeans.org.

https://github.com/apache/netbeans/assets/5039496/668926ea-dbce-4b16-b3d7-eaeeb907473f

**Comment by riksoft** [comment]
In this example wouldn't be a problem to split the text manually to fit a decent width because it's only 3 lines, but very often I have to do this for descriptions that are 200 lines, and it's a waste of time to do it manually. In Geany you only have to select the text and press CTRL-J and it is adjusted to fit the maximum width set in the options.

**Reply by Chris2011** [reply]
So you want to respect the max line width for formatting?

**Comment by riksoft** [comment]
Yes. A function to split text so that it respect the max line width.
This function should be separated from the format function (ALT+Shift+F) used for the code (and the already existing "wrap" option), since splitting code automatically I think should be avoided (it's too difficult, it involves writing tons of rules for every kind of language).

**Reply by Chris2011** [reply]
We have the wrap function and it is kind of formatting, but it just says, "none, after word or always". In general it should a new option "on max line width". Can't remember how the thing is called. But I guess you know what I mean.

**Reply by Chris2011** [reply]
Unfortunately this is not implemented and yes we should definitely need to have some.

**Comment by riksoft** [comment]
The wrap option you mention, in my case is disabled because I don't want the format function to also wrap code: it makes disasters.
